### ISSUE TITLE
Shared module gridOptions seetings not preserve in page module

### ISSUE THREAD

**Issue by lajith111** [ghiscoding/Angular-Slickgrid#1329]
### Describe the bug

I have defined grid options in shared modules
```
 AngularSlickgridModule.forRoot({
      externalResources: [new ExcelExportService()],   
      enableAutoResize: true,
      enableGridMenu: true,
      enableExcelExport:true,
      autoHeight: true,
      alwaysShowVerticalScroll: true,
}
```
In page component also some configuration defined

```
 this.gridOptions = {
      // enableExcelExport: true,  
      // externalResources: [new ExcelExportService()],    //now this needs to defined in each page after update new version
      excelExportOptions:{
        sanitizeDataExport: true,
         filename:'BankReport_'+ new Date().toJSON().slice(0,10),
         exportWithFormatter:true,
       },
```

other wise i am getting below error. 

_[Slickgrid-Universal] You must register the ExcelExportService to properly use Export to Excel in the Grid Menu. Example:: this.gridOptions = { enableExcelExport: true, externalResources: [new ExcelExportService()] };
executeGridMenuInternalCustomCommands @ slickGridMenu.js:651_

After i upgrade to 7.x i am getting this error 

if i include externalResources in  each component excel download. Thanks for the update 

i dont no this is a issue. But i found this in my testing ..

### Reproduction

After upgrading to v 7 i found this issue

### Expectation

Preserve shared module settings if not override in page module

### Environment Info

```shell
Angular 17
Angularslickgrid 7
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Quick Start](https://ghiscoding.gitbook.io/angular-slickgrid/getting-started/quick-start).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/Angular-Slickgrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/Angular-Slickgrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by ghiscoding** [comment]
ah right, I changed how external are registered in PR #1320 to avoid the `externalResources` option to no longer call the internal `mergeGridOptions` which creates a deep copy of the grid options (creating a possible duplicate instance ref) but I forgot to test it at a global level. I'll fix that later today, thanks for the early feedback

https://github.com/ghiscoding/Angular-Slickgrid/blob/0654550b02c71b2fab16c7577efa6cdee7caaf75/src/app/modules/angular-slickgrid/components/angular-slickgrid.component.ts#L1205-L1213

**Comment by ghiscoding** [comment]
fixed in v7.0.1 and v6.6.6

**Comment by lajith111** [comment]
Thanks for the quick fix..what is observed that now   working fine with externalResources but  not preserve the value of **_enableExcelExport:true,_**

export is working only if i write  enableExcelExport:true, in every component..

**Comment by ghiscoding** [comment]
Sorry I'm not sure what you said, can you provide example so I could understand better?

**Comment by lajith111** [comment]
sory just now rechecked the issue..i observed that first i downloaded excel from USerList page its download, again i goes to CustomerList page then i downloaded agian and same error You must register the ExcelExportService
Again i goes to USer List then i also get same error message

**Comment by ghiscoding** [comment]
I reverted some changes in PR #1331 but did not publish yet, let me try to publish it and hopefully that fixes your problem. Let me do that now

**Comment by ghiscoding** [comment]
please try v7.0.2 and hope it fixes it

**Comment by lajith111** [comment]
Its now working fine.. Thank you for quick fix

**Comment by ghiscoding** [comment]
Great thanks for confirming :)

**Comment by lajith111** [comment]
@ghiscoding  after i host to shared server i am getting
 **_main.42ddf1a2709f25d1.js:1  ERROR TypeError: Cannot read properties of undefined (reading 'extensionService')_**

locally using ng s it is working fine..First time i getting this error.. I am not expertize in JavaScript
i am sure that it is because of slickgrid..  do you have any idea..Thanks

**Comment by ghiscoding** [comment]
I cannot help without a full reproduction. You should try to click on the link on the errors to see where the error is thrown. 

Also please do not ask question or issues that are unrelated to the original issue.

**Comment by lajith111** [comment]
sory for that,..i thought issue in @slickgrid-universal service any way thanks for the reply...i will check

**Comment by ghiscoding** [comment]
If I cannot reproduce, then I cannot help, which is why I'm asking for a reproduction. There is not enough information to help. 

Also the GitHub Live Demo is a production build that is executed on every new release to make sure there is no error with the build, it is also used to run the 600 Cypress E2E tests... so I cannot reproduce your issue, so I cannot help

**Comment by lajith111** [comment]
now i have checked using http-server for locally to test (**ng build**) ..i find that same issue exist in local system. and using **ng s** command there is no issue. if you dont mind is there any changes related to ng build in the this library..

![image](https://github.com/ghiscoding/Angular-Slickgrid/assets/28708173/e124e222-0979-44d3-ab80-d8ea088243ce)

**Comment by lajith111** [comment]
ok..thanks i have  raised a question in  https://stackoverflow.com/questions/77690221/angular-slick-grid-error-after-deployed-package

**Comment by ghiscoding** [comment]
The issue you're facing is the same as another opened issue #1334 

It might be caused by the external dependency `multiple-select-vanilla` that I also maintain and that I started using the code `Boolean(..)` to convert input to a boolean. I pushed a new version `1.1.2` of that lib yesterday and hopefully that fixes the bug, but I have no confirmation yet if that fixed the issue or not. I will push a new version of Angular-Slickgrid today or tomorrow that will include this code change and again hoping that it closes the problem.
